src/logging.cpp

```diff
@@ -190,29 +197,37 @@ std::string BCLog::Logger::LogTimestampStr(const std::string &str)
     } else
         strStamped = str;
 
-    if (!str.empty() && str[str.size()-1] == '\n')
-        m_started_new_line = true;
-    else
-        m_started_new_line = false;
-
     return strStamped;
 }
 
+std::string BCLog::Logger::LogThreadnameStr(const std::string &str)
```

Comment:
Now that we've reversed the order of threadname and log timestamp (threadname follows timestamp), I've only modified `LogTimestampStr` per your suggestion since it's the only function of the two to receive an rvalue as input. `LogThreadnameStr` receives an lvalue from `LogPrintStr` which itself receives an lvalue from `logging.h:LogPrintf`; I may be mistaken here, but the optimization you describe wouldn't apply in this case.